Signal/src/ViewControllers/ConversationView/Cells/OWSMessageFooterView.m

```diff
@@ -141,13 +144,32 @@ - (void)animateSpinningIcon
     [self.statusIndicatorImageView.layer addAnimation:animation forKey:@"animation"];
 }
 
+- (BOOL)isFailedOutgoingMessage:(ConversationViewItem *)viewItem
+{
+    OWSAssert(viewItem);
+
+    if (viewItem.interaction.interactionType != OWSInteractionType_OutgoingMessage) {
+        return NO;
+    }
+
+    TSOutgoingMessage *outgoingMessage = (TSOutgoingMessage *)viewItem.interaction;
+    MessageReceiptStatus messageStatus =
+        [MessageRecipientStatusUtils recipientStatusWithOutgoingMessage:outgoingMessage];
+    return messageStatus == MessageReceiptStatusFailed;
+}
+
 - (void)configureLabelsWithConversationViewItem:(ConversationViewItem *)viewItem
 {
     OWSAssert(viewItem);
 
     [self configureFonts];
 
-    self.timestampLabel.text = [DateUtil formatTimestampAsTimeShort:viewItem.interaction.timestamp];
+    if ([self isFailedOutgoingMessage:viewItem]) {
+        self.timestampLabel.text
```

Comment:
maybe: s/timestampLabel/deliveryStatusLabel?

Reply:
It's a tough choice, but I'm inclined to leave it since this label will have a timestamp 99% of teh time - the "sender failure" case is the only exception.